runtime/callback.c

```diff
@@ -68,6 +68,14 @@ Caml_inline void restore_stack_parent(caml_domain_state* domain_state,
   Stack_parent(domain_state->current_stack) = parent_stack;
 }
 
+/* support code for caml_result */
+CAMLexport value caml_run_result(caml_result res)
```

Comment:
I considered the following naming choices:

1. `caml_raise_if_exception` is only in CAML_INTERNALS, so I could reuse that name and use `caml_raise_if_encoded_exception` for the old version, which is now only used in `callback.c`.
2. I could use consistent names for both versions, for example `caml_raise_if_exception_{encoded,result}`
3. or a new name altogether, for example `caml_get_value_or_raise`.

My current preference is (3) -- I like that the name is explicit that it can raise (this is better than `caml_run_result`), but also that it returns the value otherwise (this is better than `caml_raise_if_exception`).

I will move `caml_raise_if_encoded_exception` out of `fail.h` and into `callback.c` right away, and I think I will go for (3) as well, but I am interested in feedback on the name in any case.